### DISCUSSION TITLE
Support the final compilation of variables into CSS custom properties？

### DISCUSSION THREAD

**Discussion by abitwhy** [stylus/stylus#2644] [Ideas]
What if Stylus support the eventual compilation of variables into [CSS custom properties](https://developer.mozilla.org/zh-CN/docs/Web/CSS/Using_CSS_custom_properties)? Change the relationship from  competition to win-win between Stylus and CSS custom properties?

**Comment by iChenLei** [comment]
![2021-12-01 18 50 24](https://user-images.githubusercontent.com/14012511/144221323-cffdfcf9-9d07-4b42-ba1e-32e02671f593.png)

What's wrong when you use `stylus` with `css variables` ? @abitwhy

**Reply by abitwhy** [reply]
I certainly know that `CSS variables` and all other features of CSS can be used in Stylus. But what I want to emphasize is that **`Stylus variables` can eventually be compiled into `CSS variables`**, which means that Stylus developers use `Stylus variables` as `CSS variables` but will not have to write ` -- ` prefix and `var()` function when referencing. More importantly, as mentioned above, this changes the relationship between `Stylus variables` and `CSS variables`.
